src/middleware/authMiddleware.js

```diff
@@ -0,0 +1,48 @@
+const { errorResponse } = require("../helpers");
+const jwt = require("jsonwebtoken");
+const User = require("../models/userModel");
+const asyncHandler = require("express-async-handler");
+
+const loggedIn = asyncHandler(async (req, res, next) => {
+  const authHeader = req.headers.authorization;
+
+  if (!authHeader) {
+    errorResponse(res, 403, "Invalid auth token");
```

Comment:
It seems that the error code for invalid credentials should be 401. This can be aplied to line 19, 27, 32 and 39.
**403** https://www.rfc-editor.org/rfc/rfc7231#section-6.5.3
**401** https://www.rfc-editor.org/rfc/rfc7235#section-3.1